src/aotcompile.cpp

```diff
@@ -415,8 +414,8 @@ void *jl_create_native(jl_array_t *methods, const jl_cgparams_t cgparams, int _p
     }
 
     data->M = std::move(clone);
-    if (jl_measure_compile_time[tid])
-        jl_cumulative_compile_time[tid] += (jl_hrtime() - compiler_start_time);
+    if (jl_atomic_load(&jl_measure_compile_time))
+        jl_atomic_fetch_add(&jl_measure_compile_time, (jl_hrtime() - compiler_start_time));
```

Comment:
The `jl_atomic_load` should only appear exactly once in this function, or it is not actually atomic (just data-race free).

Reply:
(it could also be `jl_atomic_load_relaxed`, though the performance different shouldn't be measurable)

Reply:
Though it is actually unclear to me why this function counts against the compile-time. It is only used by the GPU stack, and is therefore not quite JIT time.

Reply:
> The `jl_atomic_load` should only appear exactly once in this function, or it is not actually atomic (just data-race free).

Ah, thanks @vtjnash! Great catch. Thanks for the careful review 👍 I'll apply this change (and those suggested throughout), and ping you again. 👍 👍

Reply:
> Though it is actually unclear to me why this function counts against the compile-time. It is only used by the GPU stack, and is therefore not quite JIT time.

Regarding this part, I'd like to leave this change out of this PR. This is only supposed to be a bug fix. Can we start a separate discussion, maybe in an issue, about this?

Reply:
I've opened https://github.com/JuliaLang/julia/issues/41921 to discuss the remaining question here. I'll now resolve this thread.